Userland/Applications/Assistant/Providers.cpp

```diff
@@ -48,15 +49,7 @@ void FileResult::activate() const
 
 void TerminalResult::activate() const
 {
-    pid_t pid;
-    char const* argv[] = { "Terminal", "-k", "-e", title().characters(), nullptr };
-
-    if ((errno = posix_spawn(&pid, "/bin/Terminal", nullptr, nullptr, const_cast<char**>(argv), environ))) {
-        perror("posix_spawn");
-    } else {
-        if (disown(pid) < 0)
-            perror("disown");
-    }
+    (void)Core::Process::spawn("/bin/Terminal", Array { "-k", "-e", title().characters() });
```

Comment:
This doesn't do what the commit message claims it would do :^)

Reply:
Why isn't it using `spawn_or_show_error()` tho? Assistant is a GUI app.

Reply:
Because it's a an `Assistant::Result` object rather than a `GUI::Widget` so it does not have access to the applications window, and it would take a fair bit of finagling to get the `GUI::Window*` pointer there I think. 

I believe you would have to pass it to the `Database` object, & then make that pass it to the `TerminalProvider`, & then make that pass it to the `TerminalResult`.

Reply:
Slap a FIXME on it and I'll take it with the current commit message :)

Reply:
(you also also maybe stick the Window* in a global & make a getter for it, but that's a little ugly just for this)